### PR TITLE
Feature/675

### LINKED ISSUE #675: As a citizen I would like to be able to mark an activity as complete directly from the week plan
## Description
**Describe the Task** 

As a citizen, I should be able to mark an activity as complete from the week plan, so I don't need to access the activity screen. I should also be able to start an activity with a timer from the week plan, and it should auto-complete when the timer runs out. 

**Possible Suggested Solution**  
Change so the citizen is not able to access the activity page anymore. Instead, they will look at activities in the current week plan. They will, from here, be able to mark activities as complete and start timers which will auto-complete when the time runs out. 

Potentially in the future, add an option that allows a guardian to decide if a citizen should be able to access the activity screen.

### PR DESCRIPTION
**Pull request by manfredhod**
# Description

Adds the new feature where the citizen now marks an activity as complete from the week plan itself. Furthermore, the citizen will start the timer from the week plan, which will autocomplete when it runs out. The Guardian can access the activity screen as before, and the timer will also autocomplete when it runs out. 

Fixes #675

Related pull requests:
https://github.com/aau-giraf/api_client/pull/121
https://github.com/aau-giraf/web-api/pull/298

## Type of change
*Delete unchecked boxes (only for Type of change)*

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The new feature has been tested on a physical device and on the emulator in Android Studio.

**Development Configuration**
*Type "flutter --version" and "dart --version" in your CMD to check versions.*

* Flutter version: 3.3.8
* Dart version: 2.18.4

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [x] I have Acceptance Tested this on an Android device

### CODE REVIEW

**Comment by manfredhod** [comment]
image
Disse errors bliver fixed når vores opdatering af api_client også bliver merged ind i develop branchen på api_client.

**Comment by manfredhod** [comment]
image
Ignore not related to our PR will be fixed when everything is merged into develop and our changes to the api_client are accepted.

**Comment by manfredhod** [comment]
Some tests might fail in GitHub actions they work locally. The reason is that the GitHub Actions tests run with flutter 1.20.x, where some tests fail randomly. The tests will work as intended when the new version of flutter is applied to GitHub Actions.

File: pubspec.yaml [hunk 1]
```diff
@@ -13,12 +13,16 @@ dependencies:
   api_client:
     git:
       url: https://github.com/aau-giraf/api_client.git
-      ref: develop
```
Comment (Just1Per) [inline pubspec.yaml:16]: Should this point to develop again, if so remember to change back :)

Comment (Just1Per) [inline pubspec.yaml:16]: Should this point to develop? if so remember to add back :)

**Review by Just1Per — DISMISSED** [review]

File: pubspec.yaml (hunk 1 above)
Comment (manfredhod) [reply pubspec.yaml:16]: We will add it back before merging :)

**Comment by J2KJonas** [comment]
image
Fix tests that do not pass?

**Comment by J2KJonas** [comment]
image
image
image
image
mangler at pull fra develop eller fikse disse fejlede test

**Review by J2KJonas — DISMISSED** [review]
image
we good now, approved

**Comment by manfredhod** [comment]
> image image image image mangler at pull fra develop eller fikse disse fejlede test

Underligt ingen af vores tests fejler og GitHub actions går igennem. Vi har også alle commits fra develop.

image

Er du sikker på du har den nyeste version?

**Comment by manfredhod** [comment]
> image we good now, approved

Super

**Review by Just1Per — DISMISSED** [review]

**Review by Just1Per — APPROVED** [review]

**Pull request merged**